src/pudl/transform/eia861.py

```diff
@@ -2327,76 +2146,290 @@ def utility_data(tfr_dfs):
         how="outer",
     )
 
-    # Drop original operational_data_eia861 table from tfr_dfs
-    del tfr_dfs["utility_data_eia861"]
-
-    tfr_dfs["utility_data_nerc_eia861"] = transformed_ud_nerc
-    tfr_dfs["utility_data_rto_eia861"] = transformed_ud_rto
-    tfr_dfs["utility_data_misc_eia861"] = transformed_ud_misc
-
-    return tfr_dfs
+    return (
+        Output(
+            output_name="utility_data_nerc_eia861",
+            value=_post_process(transformed_ud_nerc),
+        ),
+        Output(
+            output_name="utility_data_rto_eia861",
+            value=_post_process(transformed_ud_rto),
+        ),
+        Output(
+            output_name="utility_data_misc_eia861",
+            value=_post_process(transformed_ud_misc),
+        ),
+    )
 
 
 ##############################################################################
-# Coordinating Transform Function
+# Secondary Transformations
 ##############################################################################
+@asset(
+    ins={
+        "advanced_metering_infrastructure_eia861": AssetIn(),
+        "demand_response_eia861": AssetIn(),
+        "demand_response_water_heater_eia861": AssetIn(),
+        "demand_side_management_ee_dr_eia861": AssetIn(),
+        "demand_side_management_misc_eia861": AssetIn(),
+        "demand_side_management_sales_eia861": AssetIn(),
+        "distributed_generation_fuel_eia861": AssetIn(),
+        "distributed_generation_misc_eia861": AssetIn(),
+        "distributed_generation_tech_eia861": AssetIn(),
+        "distribution_systems_eia861": AssetIn(),
+        "dynamic_pricing_eia861": AssetIn(),
+        "energy_efficiency_eia861": AssetIn(),
+        "green_pricing_eia861": AssetIn(),
+        "mergers_eia861": AssetIn(),
+        "net_metering_customer_fuel_class_eia861": AssetIn(),
+        "net_metering_misc_eia861": AssetIn(),
+        "non_net_metering_customer_fuel_class_eia861": AssetIn(),
+        "non_net_metering_misc_eia861": AssetIn(),
+        "operational_data_misc_eia861": AssetIn(),
+        "operational_data_revenue_eia861": AssetIn(),
+        "reliability_eia861": AssetIn(),
+        "sales_eia861": AssetIn(),
+        "utility_data_misc_eia861": AssetIn(),
+        "utility_data_nerc_eia861": AssetIn(),
+        "utility_data_rto_eia861": AssetIn(),
+    },
```

Comment:
I tried and failed to find an easy way to select these assets dynamically, but it seems like that should be possible. What's the right way to do it? This is a large majority of the assets defined in this module.  I thought I could do it by assigning them to different asset groups, but then I couldn't figure out how to get the asset ID based on the asset groups. I could probably load all of the assets defined in this module and then exclude the smaller number that shouldn't be inputs here, but that still feels brittle, and it seems like there should be some way to specify selectable metadata on the assets when they're defined, so they can be selected later for particular purposes.

Reply:
You could also create a new module for these mutli_assets that depends on upstream eia861 asset groups.

Reply:
There's also `load_assets_from_current_module()` but I wasn't sure if that would result in the asset depending on itself, which isn't right.

This also isn't *all* the assets. It excludes a couple that are structural rather than data tables.

Maybe once we get all the outputs converted to Dagster we should talk about re-organizing the modules / filesystem hierarchy to resemble Dagster's model project layout? With a top level assets directory, and several "thematic subdirectories inside it, e.g. one for each data source (`eia860`, `eia923`, `epacems` etc.) as well as each bundle of related analyses (`mcoe`, `state_demand`, `depreciation` etc.).